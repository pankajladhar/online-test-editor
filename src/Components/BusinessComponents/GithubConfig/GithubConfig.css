.GithubConfig {
  padding-top: 20px; }
  .GithubConfig h2 {
    margin-bottom: 0px;
    text-align: center; }
  .GithubConfig h3, .GithubConfig h4 {
    margin-bottom: 0; }
  .GithubConfig img {
    width: 100%;
    display: block;
    margin-bottom: 20px;
    border: 1px #1c2234 solid; }
  .GithubConfig .GithubConfig__FileName {
    font-weight: bold;
    text-decoration: underline;
    display: inline-block;
    padding: 0 3px; }
  .GithubConfig .btn {
    margin-top: 10px;
    display: inline-block; }
  .GithubConfig .GithubConfig__Info {
    display: flex;
    justify-content: space-between; }
