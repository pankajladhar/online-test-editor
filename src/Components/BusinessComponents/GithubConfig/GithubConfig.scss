.GithubConfig {
  padding-top: 20px;
  h2 {
    margin-bottom: 0px;
    text-align: center;
  }

  h3, h4 {
    margin-bottom: 0;
  }

  img {
    width: 100%;
    display: block;
    margin-bottom: 20px;
    border: 1px #1c2234 solid;
  }

  .GithubConfig__FileName {
    font-weight: bold;
    text-decoration: underline;
    display: inline-block;
    padding: 0 3px;
  }
}