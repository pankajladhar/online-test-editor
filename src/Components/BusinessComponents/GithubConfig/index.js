import React, { Component } from 'react';
import PropTypes from 'prop-types';
import './GithubConfig.css'
import step1 from './images/step-1.png';
import step2 from './images/step-2.png';
import step3 from './images/step-3.png';

class GithubConfig extends Component {
    render() {
        return (
            <div className="GithubConfig">
                <h2>How to configure using github</h2>
                <div className="GithubConfig__Info">
                    <div>
                        <h3> Assumption :</h3>
                        You have pushed
                        <span className="GithubConfig__FileName">react_new.json</span>
                        file in one of your repository.
                        <div>
                            * extension of file must be <span className="GithubConfig__FileName">.json</span>
                            and validate json.
                        </div>
                    </div>
                    <div>
                        <a className="btn" href='/Sample.json' download>Download Sample</a>
                    </div>
                </div>

                <h3>Step 1</h3>
                <img src={step1} alt="step1" />
                <h3>Step 2</h3>
                <img src={step2} alt="step2" />
                <h3>Step 3</h3>
                <img src={step3} alt="step3" />
            </div>
        );
    }
}

GithubConfig.propTypes = {

};

export default GithubConfig;