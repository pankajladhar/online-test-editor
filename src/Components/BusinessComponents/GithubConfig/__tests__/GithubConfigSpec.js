import React from 'react';
import renderer from 'react-test-renderer';
// import { shallow } from 'enzyme';
import GithubConfig from './../index';

describe('GithubConfig', ()=>{
    it('renders correctly', ()=>{
        let tree = renderer.create(
            <GithubConfig/>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});