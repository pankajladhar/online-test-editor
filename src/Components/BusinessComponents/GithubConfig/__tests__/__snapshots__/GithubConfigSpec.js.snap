// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GithubConfig renders correctly 1`] = `
<div
  className="GithubConfig"
>
  <h2>
    How to configure using github
  </h2>
  <div
    className="GithubConfig__Info"
  >
    <div>
      <h3>
         Assumption :
      </h3>
      You have pushed
      <span
        className="GithubConfig__FileName"
      >
        react_new.json
      </span>
      file in one of your repository.
      <div>
        * extension of file must be 
        <span
          className="GithubConfig__FileName"
        >
          .json
        </span>
        and validate json.
      </div>
    </div>
    <div>
      <a
        className="btn"
        download={true}
        href="/Sample.json"
      >
        Download Sample
      </a>
    </div>
  </div>
  <h3>
    Step 1
  </h3>
  <img
    alt="step1"
    src="step-1.png"
  />
  <h3>
    Step 2
  </h3>
  <img
    alt="step2"
    src="step-2.png"
  />
  <h3>
    Step 3
  </h3>
  <img
    alt="step3"
    src="step-3.png"
  />
</div>
`;
